src/util/error.cpp

```diff
@@ -8,14 +8,53 @@ extern "C" {
 TError::TError() : Error(EError::Success) {
 }
 
-TError::TError(EError e, std::string description, int eno) :
+TError::TError(EError e, const std::string &description, int eno) :
     Error(e), Description(description), Errno(eno) {
 }
 
-TError::TError(EError e, int eno, std::string description) :
+TError::TError(EError e, int eno, const std::string &description) :
     Error(e), Description(std::string(strerror(eno)) + ": " + description), Errno(eno) {
 }
 
+TError::TError(EError e, std::string &&description, int eno) :
+    Error(e), Description(std::move(description)), Errno(eno) {
+}
+
+TError::TError(const TError &other) :
```

Comment:
Why do you need explicit copy and move constructors while compiler would generate them implicitly just in the way they are implemented here? If you do want to explicitly define a ctor you can use something like
'TError(TError&&) = default' in header file